internal/worker/worker.go

```diff
@@ -152,12 +162,11 @@ func (w *worker) startJobLoop(ctx context.Context) <-chan error {
 
 			if job != nil {
 				eg.Go(safety.RecoverFunc(func() (err error) {
-					defer atomic.AddUint64(&w.completedCount, 1)
-					err = job(ctx)
-					if err != nil {
+					if err = job(ctx); err != nil {
 						log.Debugf("an error occurred while executing a job: %s", err)
 						ech <- err
 					}
+					atomic.AddUint64(&w.completedCount, 1)
```

Comment:
I think defer is better if ech was closed, it makes runtime panic so counter is not increase even jos finished